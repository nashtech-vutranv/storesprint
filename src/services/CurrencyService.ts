import {AxiosInstance} from 'axios'

class CurrencyService {
  private axiosClient

  constructor(axiosClient: AxiosInstance) {
    axiosClient.defaults.baseURL = process.env.REACT_APP_API_SERVER_URL
    this.axiosClient = axiosClient
  }

  fetchAllCurrencies() {
    const aggregatorsData = this.axiosClient.get('/currencies')
    return aggregatorsData
  }
}

export default CurrencyService
