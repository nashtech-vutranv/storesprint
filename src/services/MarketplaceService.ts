import {AxiosInstance} from 'axios'
import {
  generateFilterPaginationQuery,
  buildPaginationQueryOpts,
} from '../helpers/queryParams'
import {
  IUnAssignedRequestBodyMarketplace,
  IBodyMarketplace,
  IMarketplaceFormRequest,
} from '../interface/marketplace'
import {IPagination} from '../interface/pagination'

class MarketplaceService {
  private axiosClient

  constructor(axiosClient: AxiosInstance) {
    axiosClient.defaults.baseURL = process.env.REACT_APP_API_SERVER_URL
    this.axiosClient = axiosClient
  }

  fetchAllMarketplaces(
    paginationQuery: Omit<IPagination, 'keyword'> | null,
    searchObj: {search: string; requiredHasAggregator?: boolean}
  ) {
    const marketplacesData = this.axiosClient.post(
      `/marketplaces/filter?${generateFilterPaginationQuery(
        paginationQuery ?? undefined
      )}`,
      {
        search: searchObj.search,
        requiredHasAggregator: Boolean(searchObj.requiredHasAggregator),
      }
    )
    return marketplacesData
  }

  fetchMarketplaceById(id?: string) {
    return this.axiosClient.get(`/marketplaces/${id}`)
  }

  getUnAssignedMarketplaces(entity: IUnAssignedRequestBodyMarketplace) {
    entity.requiredHasAggregator = Boolean(entity.requiredHasAggregator)
    return this.axiosClient.post('/marketplaces/unassigned/filter', entity)
  }

  addMarketplace(entity: IMarketplaceFormRequest) {
    return this.axiosClient.post('/marketplaces', entity)
  }

  editMarketplace(entity: IMarketplaceFormRequest) {
    return this.axiosClient.put(`/marketplaces/${entity.id}`, entity)
  }

  getMarketplacesWithIsProductReviewProp(
    paginationQueryParam: any,
    entity: IBodyMarketplace
  ) {
    return this.axiosClient.post(
      `/marketplaces/filter?${buildPaginationQueryOpts(paginationQueryParam)}`,
      entity
    )
  }

  getMarketplaceTypeInformation() {
    return this.axiosClient.get('/marketplace-types', {
      baseURL: process.env.REACT_APP_API_MARKETPLACE_SERVER_URL
    })
  }
}

export default MarketplaceService
