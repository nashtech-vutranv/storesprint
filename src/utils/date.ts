import moment from 'moment'
import {FORMAT_DATE, FORMATE_DATE_NO_HOURS} from '../constants'

export const formatDate = (date: string, format: string = FORMAT_DATE) => {
  return moment(date).format(format)
}

export const formatDateNoHours = (date: string, format: string = FORMATE_DATE_NO_HOURS) => {
  return moment(date).format(format)
}
