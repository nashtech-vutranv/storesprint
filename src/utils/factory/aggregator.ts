import faker from '@faker-js/faker'

export const mockAggregator = () => {
  return {
    id: faker.random.word(),
    name: faker.random.word(),
    version: faker.datatype.number(),
    status: faker.datatype.string(),
  }
}

export const mockAggregators = () => {
  return {
    data: {
      content: Array.apply(null, Array(10)).map(() => mockAggregator()),
      totalElements: 10,
    },
  }
}
