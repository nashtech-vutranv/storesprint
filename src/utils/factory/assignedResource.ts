import faker from '@faker-js/faker'

export const mockAssignedResource = () => {
  return {
    data: {
      createdAt: faker.datatype.string(),
      id: faker.datatype.string(),
      modifiedAt: faker.datatype.string(),
      organizationResponse: {
        additionalInformation: faker.datatype.string(),
        addressLine1: faker.datatype.string(),
        addressLine2: null,
        addressLine3: null,
        city: faker.datatype.string(),
        contactEmailAddress: faker.datatype.string(),
        contactName: faker.datatype.string(),
        contactPhoneNumber: faker.datatype.string(),
        createdAt: faker.datatype.string(),
        erpId: faker.datatype.string(),
        id: faker.datatype.string(),
        modifiedAt: faker.datatype.string(),
        name: faker.datatype.string(),
        postCode: faker.datatype.string(),
        registrationNumber: faker.datatype.string(),
        status: faker.datatype.string(),
        vatRegistrationNumber: faker.datatype.string(),
        version: faker.datatype.number(),
      },
      siteResponse: {
        createdAt: faker.datatype.string(),
        erpId: faker.datatype.string(),
        id: faker.datatype.string(),
        modifiedAt: faker.datatype.string(),
        name: faker.datatype.string(),
        organizationId: faker.datatype.string(),
        status: faker.datatype.string(),
        url: faker.datatype.string(),
        version: faker.datatype.number(),
      },
      userResponse: null,
      version: faker.datatype.number(),
    },
  }
}
