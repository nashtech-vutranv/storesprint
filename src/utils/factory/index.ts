export * from './organization'
export * from './warehouse'
export * from './aggregator'
export * from './marketplaces'
export * from './currency'
export * from './relationshipAM'
export * from './marketplaceRelationships'
export * from './product'
export * from './order'
export * from './listingStatus'
export * from './common'
export * from './defaultPropertyValue'
export * from './mappings'
export * from './productProperties'
export * from './assignedResource'
export * from './productCategoryMappings'
export * from './productPropertyMappings'
export * from './courier'
export * from './valueMapping'
export * from './courierMappings'
export * from './returnsrefunds'
export * from './deliveryTypeMappings'
