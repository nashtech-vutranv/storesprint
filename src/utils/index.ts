export * from './array'
export * from './layout'
export * from './date'
export * from './conditionPromise'
