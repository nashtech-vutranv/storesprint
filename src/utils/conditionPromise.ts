import {mockOrganizations, mockSites, mockSiteLocales} from './factory/organization'
import {mockMarketplaces, mockCalculateAssignMarketplaces} from './factory/marketplaces'
import {mockAssignedProducts, mockAssignedProductsResponse} from './factory/product'
import {mockWarehouses} from './factory/warehouse'
import {mockMarketplaceRelationship} from './factory/marketplaceRelationships'

type IMockType = 'organization' | 'site' | 'siteLocale' |
 'marketplace' | 'calculateMarketplace' | 'warehouse' 
 | 'marketplaceRelationship' | 'assignedProduct' | 'assignedProductResponse'

export const conditionPromise = (objectFilter: unknown, typeMock: IMockType, objectPaginationQuery?: any) => {
  if (objectFilter || (objectFilter && objectPaginationQuery)) {
    switch (typeMock) {
      case 'organization': return Promise.resolve(mockOrganizations())
      case 'site': return Promise.resolve(mockSites())
      case 'siteLocale': return Promise.resolve(mockSiteLocales())
      case 'marketplace': return Promise.resolve(mockMarketplaces())
      case 'calculateMarketplace': return Promise.resolve(mockCalculateAssignMarketplaces())
      case 'warehouse': return mockWarehouses()
      case 'marketplaceRelationship': return mockMarketplaceRelationship()
      case 'assignedProduct': return Promise.resolve(mockAssignedProducts())
      case 'assignedProductResponse': return Promise.resolve(mockAssignedProductsResponse())

      default: return {}
    }
  } else return Promise.reject()
}