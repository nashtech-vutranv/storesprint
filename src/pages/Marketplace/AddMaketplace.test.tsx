import axios from 'axios'
import {fireEvent, render, waitFor} from '@testing-library/react'
import '@testing-library/jest-dom'
import {MemoryRouter} from 'react-router-dom'
import MarketplaceService from '../../services/MarketplaceService'
import CurrencyService from '../../services/CurrencyService'
import {mockMarketplace} from '../../utils/factory'
import {UpdateMarketplaces} from '.'

const tree = render(
    <MemoryRouter>
      <UpdateMarketplaces />
    </MemoryRouter>
  )

describe('Add Marketplace Page', () => {
  beforeEach(() => {
    jest.mock('react-router-dom', () => {
      return {
        ...jest.requireActual('react-router-dom'),
        useParams: () => ({marketplaceId: undefined}),
      }
    })
  })

  it('should match snapshot with successful data from api', async () => {
    jest
      .spyOn(new CurrencyService(axios) as any, 'fetchAllCurrencies')
      .mockImplementation((currenciesData) => {
        if (currenciesData) {
          return currenciesData
        } else return Promise.reject()
      })

    jest
      .spyOn(new MarketplaceService(axios) as any, 'addMarketplace')
      .mockImplementation((marketplaceId) => {
        if (marketplaceId !== 'existed_marketplace_id') {
          return Promise.resolve({data: mockMarketplace()})
        } else return Promise.reject()
      })

    await waitFor(() => {
      expect(tree).toMatchSnapshot()
      expect(tree.getByText(/marketplace_add_title/i)).toBeInTheDocument()
      expect(tree.getByText(/marketplace_details_name/i)).toBeInTheDocument()
      expect(
        tree.getByText(/marketplace_details_status_active/i)
      ).toBeInTheDocument()
    })
    
    const currencySelect = tree.container.querySelectorAll(
      'input[role="combobox"]'
    )[0] as Element

    fireEvent.change(currencySelect, {target: {value: 'cur-selected'}})
    fireEvent.blur(currencySelect)

    const saveBtn = tree.getByText(/common_confirm_save/i)
    fireEvent.click(saveBtn)
    const cancelBtn = tree.getByText(/common_confirm_cancel/i)
    fireEvent.click(cancelBtn)
  })
})
