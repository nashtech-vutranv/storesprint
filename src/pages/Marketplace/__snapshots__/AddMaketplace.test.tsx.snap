// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Marketplace Page should match snapshot with successful data from api 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-form mt-3 card"
      >
        <div
          class="card-header"
        >
          <h4
            class="card-form__title"
          >
            marketplace_add_title
          </h4>
        </div>
        <div
          class="card-body"
        >
          <div
            class="row"
          >
            <form
              autocomplete="on"
              class="form-layout"
            >
              <div
                class="container-fluid"
              >
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class="required"
                      for="marketplaceTypeId"
                    >
                      marketplace_details_type
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <div
                      class="react-select inherit-color css-b62m3t-container"
                      id="marketplaceTypeId"
                    >
                      <span
                        class="css-1f43avz-a11yText-A11yText"
                        id="react-select-2-live-region"
                      />
                      <span
                        aria-atomic="false"
                        aria-live="polite"
                        aria-relevant="additions text"
                        class="css-1f43avz-a11yText-A11yText"
                      />
                      <div
                        class="react-select__control css-13cymwt-control"
                      >
                        <div
                          class="react-select__value-container css-1fdsijx-ValueContainer"
                        >
                          <div
                            class="react-select__placeholder css-1jqq78o-placeholder"
                            id="react-select-2-placeholder"
                          >
                            marketplace_placeholder_type
                          </div>
                          <div
                            class="react-select__input-container css-qbdosj-Input"
                            data-value=""
                          >
                            <input
                              aria-autocomplete="list"
                              aria-describedby="react-select-2-placeholder"
                              aria-expanded="false"
                              aria-haspopup="true"
                              aria-label="textbox"
                              autocapitalize="none"
                              autocomplete="off"
                              autocorrect="off"
                              class="react-select__input"
                              id="react-select-2-input"
                              role="combobox"
                              spellcheck="false"
                              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                              tabindex="0"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                        <div
                          class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                        >
                          <span
                            class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
                          />
                          <div
                            aria-hidden="true"
                            class="react-select__indicator react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
                          >
                            <svg
                              aria-hidden="true"
                              class="css-tj5bde-Svg"
                              focusable="false"
                              height="20"
                              viewBox="0 0 20 20"
                              width="20"
                            >
                              <path
                                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                              />
                            </svg>
                          </div>
                        </div>
                      </div>
                      <input
                        name="marketplaceTypeId"
                        type="hidden"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class="required"
                      for="name"
                    >
                      marketplace_details_name
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      class="p-inputtext p-component w-full p-1"
                      id="name"
                      maxlength="128"
                      name="name"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="row"
                />
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class="required"
                      for="currencyId"
                    >
                      marketplace_details_currency
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <div
                      class="react-select inherit-color css-b62m3t-container"
                      id="currencyId"
                    >
                      <span
                        class="css-1f43avz-a11yText-A11yText"
                        id="react-select-3-live-region"
                      />
                      <span
                        aria-atomic="false"
                        aria-live="polite"
                        aria-relevant="additions text"
                        class="css-1f43avz-a11yText-A11yText"
                      />
                      <div
                        class="react-select__control css-13cymwt-control"
                      >
                        <div
                          class="react-select__value-container css-1fdsijx-ValueContainer"
                        >
                          <div
                            class="react-select__placeholder css-1jqq78o-placeholder"
                            id="react-select-3-placeholder"
                          >
                            marketplace_placeholder_currency
                          </div>
                          <div
                            class="react-select__input-container css-qbdosj-Input"
                            data-value=""
                          >
                            <input
                              aria-autocomplete="list"
                              aria-describedby="react-select-3-placeholder"
                              aria-expanded="false"
                              aria-haspopup="true"
                              aria-label="textbox"
                              autocapitalize="none"
                              autocomplete="off"
                              autocorrect="off"
                              class="react-select__input"
                              id="react-select-3-input"
                              role="combobox"
                              spellcheck="false"
                              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                              tabindex="0"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                        <div
                          class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                        >
                          <span
                            class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
                          />
                          <div
                            aria-hidden="true"
                            class="react-select__indicator react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
                          >
                            <svg
                              aria-hidden="true"
                              class="css-tj5bde-Svg"
                              focusable="false"
                              height="20"
                              viewBox="0 0 20 20"
                              width="20"
                            >
                              <path
                                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                              />
                            </svg>
                          </div>
                        </div>
                      </div>
                      <input
                        name="currencyId"
                        type="hidden"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class=""
                      for="isActive"
                    >
                      marketplace_details_status_active
                    </label>
                  </div>
                  <div
                    class="col-8"
                  >
                    <div
                      class="p-inputswitch p-component p-inputswitch-checked"
                    >
                      <div
                        class="p-hidden-accessible"
                      >
                        <input
                          aria-checked="true"
                          aria-label="input-checkbox"
                          checked=""
                          name="status"
                          type="checkbox"
                        />
                      </div>
                      <span
                        class="p-inputswitch-slider"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col-8 offset-4"
                  >
                    <button
                      class="me-2 btn btn-danger"
                      type="button"
                    >
                      common_confirm_cancel
                    </button>
                    <button
                      class="me-2 btn btn-success"
                      disabled=""
                      type="button"
                    >
                      common_confirm_save
                    </button>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-form mt-3 card"
    >
      <div
        class="card-header"
      >
        <h4
          class="card-form__title"
        >
          marketplace_add_title
        </h4>
      </div>
      <div
        class="card-body"
      >
        <div
          class="row"
        >
          <form
            autocomplete="on"
            class="form-layout"
          >
            <div
              class="container-fluid"
            >
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class="required"
                    for="marketplaceTypeId"
                  >
                    marketplace_details_type
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <div
                    class="react-select inherit-color css-b62m3t-container"
                    id="marketplaceTypeId"
                  >
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-2-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class="react-select__control css-13cymwt-control"
                    >
                      <div
                        class="react-select__value-container css-1fdsijx-ValueContainer"
                      >
                        <div
                          class="react-select__placeholder css-1jqq78o-placeholder"
                          id="react-select-2-placeholder"
                        >
                          marketplace_placeholder_type
                        </div>
                        <div
                          class="react-select__input-container css-qbdosj-Input"
                          data-value=""
                        >
                          <input
                            aria-autocomplete="list"
                            aria-describedby="react-select-2-placeholder"
                            aria-expanded="false"
                            aria-haspopup="true"
                            aria-label="textbox"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            class="react-select__input"
                            id="react-select-2-input"
                            role="combobox"
                            spellcheck="false"
                            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          class="react-select__indicator react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
                        >
                          <svg
                            aria-hidden="true"
                            class="css-tj5bde-Svg"
                            focusable="false"
                            height="20"
                            viewBox="0 0 20 20"
                            width="20"
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="marketplaceTypeId"
                      type="hidden"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class="required"
                    for="name"
                  >
                    marketplace_details_name
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    class="p-inputtext p-component w-full p-1"
                    id="name"
                    maxlength="128"
                    name="name"
                    value=""
                  />
                </div>
              </div>
              <div
                class="row"
              />
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class="required"
                    for="currencyId"
                  >
                    marketplace_details_currency
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <div
                    class="react-select inherit-color css-b62m3t-container"
                    id="currencyId"
                  >
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-3-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class="react-select__control css-13cymwt-control"
                    >
                      <div
                        class="react-select__value-container css-1fdsijx-ValueContainer"
                      >
                        <div
                          class="react-select__placeholder css-1jqq78o-placeholder"
                          id="react-select-3-placeholder"
                        >
                          marketplace_placeholder_currency
                        </div>
                        <div
                          class="react-select__input-container css-qbdosj-Input"
                          data-value=""
                        >
                          <input
                            aria-autocomplete="list"
                            aria-describedby="react-select-3-placeholder"
                            aria-expanded="false"
                            aria-haspopup="true"
                            aria-label="textbox"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            class="react-select__input"
                            id="react-select-3-input"
                            role="combobox"
                            spellcheck="false"
                            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          class="react-select__indicator react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
                        >
                          <svg
                            aria-hidden="true"
                            class="css-tj5bde-Svg"
                            focusable="false"
                            height="20"
                            viewBox="0 0 20 20"
                            width="20"
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="currencyId"
                      type="hidden"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class=""
                    for="isActive"
                  >
                    marketplace_details_status_active
                  </label>
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="p-inputswitch p-component p-inputswitch-checked"
                  >
                    <div
                      class="p-hidden-accessible"
                    >
                      <input
                        aria-checked="true"
                        aria-label="input-checkbox"
                        checked=""
                        name="status"
                        type="checkbox"
                      />
                    </div>
                    <span
                      class="p-inputswitch-slider"
                    />
                  </div>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col-8 offset-4"
                >
                  <button
                    class="me-2 btn btn-danger"
                    type="button"
                  >
                    common_confirm_cancel
                  </button>
                  <button
                    class="me-2 btn btn-success"
                    disabled=""
                    type="button"
                  >
                    common_confirm_save
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
